src/docfx/build/xref/XrefMap.cs

```diff
@@ -47,170 +45,65 @@ public XrefMap(Context context, IReadOnlyDictionary<string, List<Lazy<IXrefSpec>
-                    spec.Href += internalSpec.DeclairingFile.Docset.Repository?.Branch == "master" ? "?branch=master" : "";
-                    loadedInternalSpecs.Add(spec);
-                }
-            }
-            return loadedInternalSpecs;
+            return _externalXrefMap.TryGetValue(uid, out var result) ? result.Value : null;
         }
 
-        private IXrefSpec GetLatestInternalXrefMap(List<IXrefSpec> specs)
-            => specs.SingleOrDefault(x => x.Monikers?.Any() != true)
-               ?? specs.Where(x => x.Monikers?.Any() != false).OrderByDescending(item => item.Monikers.FirstOrDefault(), _context.MonikerProvider.Comparer).FirstOrDefault();
-
-        private bool TryGetValidXrefSpecs(string uid, List<IXrefSpec> specsWithSameUid, out List<IXrefSpec> validSpecs)
+        private IXrefSpec ResolveInternalXrefSpec(string uid, string moniker)
         {
-            validSpecs = new List<IXrefSpec>();
-
-            // no conflicts
-            if (specsWithSameUid.Count() == 1)
-            {
-                validSpecs.AddRange(specsWithSameUid);
-                return true;
-            }
-
-            // multiple uid conflicts without moniker range definition, drop the uid and log an error
-            var conflictsWithoutMoniker = specsWithSameUid.Where(item => item.Monikers.Count == 0);
-            if (conflictsWithoutMoniker.Count() > 1)
+            if (!_internalXrefMap.TryGetValue(uid, out var specs))
             {
-                var orderedConflict = conflictsWithoutMoniker.OrderBy(item => item.Href);
-                _context.ErrorLog.Write(Errors.UidConflict(uid, orderedConflict.Select(x => x.DeclairingFile.FilePath)));
-                return false;
+                return null;
             }
-            else if (conflictsWithoutMoniker.Count() == 1)
-            {
-                validSpecs.Add(conflictsWithoutMoniker.Single());
-            }
-
-            // uid conflicts with overlapping monikers, drop the uid and log an error
-            var conflictsWithMoniker = specsWithSameUid.Where(x => x.Monikers.Count > 0);
-            if (CheckOverlappingMonikers(specsWithSameUid, out var overlappingMonikers))
-            {
-                _context.ErrorLog.Write(Errors.MonikerOverlapping(overlappingMonikers));
-                return false;
-            }
-
-            // define same uid with non-overlapping monikers, add them all
-            else
-            {
-                validSpecs.AddRange(conflictsWithMoniker);
-                return true;
-            }
-        }
 
-        private bool CheckOverlappingMonikers(IEnumerable<IXrefSpec> specsWithSameUid, out HashSet<string> overlappingMonikers)
-        {
-            bool isOverlapping = false;
-            overlappingMonikers = new HashSet<string>();
-            var monikerHashSet = new HashSet<string>();
-            foreach (var spec in specsWithSameUid)
+            if (!string.IsNullOrEmpty(moniker))
```

Comment:
We are not covering all the scenarios in this PR. Check https://github.com/dotnet/docfx/pull/4621.
I am scheduling a meeting to have a requirement review firstly, then continue with this.

Reply:
yep, just keep existing code untouched